code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -1107,20 +1107,28 @@ func (sc *StorageSmartContract) addChallenge(alloc *StorageAllocation,
 	}
 
 	// remove expired challenges
-	expiredIDs, err := alloc.removeExpiredChallenges(allocChallenges, challenge.Created, challenge.BlobberID)
-	lenExpiredIDs   := len(expiredIDs)
+	expiredIDsMap, totalExpiredCount, err := alloc.removeExpiredChallenges(allocChallenges, challenge.Created, challenge.BlobberID)
```

Comment:
Why not just return the map of open challenge id and blobber id,  i.e. `map[string]string` instead of using blobber id as key. The how process will be simplified, we can also check the duplication if necessary when collecting the expired challenges. And we wouldn't need the second `totalExpiredCount` returned value.